.github/workflows/post_publish.yml

```diff
@@ -51,3 +51,52 @@ jobs:
     if: ${{ github.event.workflow_run.conclusion == 'failure' }}
     steps:
       - run: echo 'The triggering workflow failed'
+
+
+  registry-check:
+    # strategy:
+    #   matrix:
+    #     os: [ubuntu-latest, windows-latest, macOS-latest]
+    # runs-on: ${{ matrix.os }}
+    # runs-on: ubuntu-latest
+    runs:
+      using: "composite"
+      steps:
+        - using: composite
+        - name: Registry Version Check
+          id: registry_latest_ver
+          shell: bash
+          run: |
+            curl -s "https://registry.terraform.io/v2/providers/323/provider-versions/latest" | jq '.data.attributes.version' >> "$GITHUB_OUTPUT"
+
+        - name: Check Version Values
+          if: ${{ needs.on-success-or-workflow-dispatch.outputs.release-tag }} == steps.registry_latest_ver.outputs.current
```

Comment:
Thinking about this a bit more, I'm not sure this `if` is going to do what we want it to after all. Currently, if `needs.on-success-or-workflow-dispatch.outputs.release-tag != steps.registry_latest_ver.outputs.current`, this step will just get skipped and the workflow will move on to the next step.

I'm kinda thinking we need to have this step run every time (no `if`), and then do a loop until the current version matches the one from